libraries/AP_EFI/AP_EFI_MAV.h

```diff
@@ -0,0 +1,32 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "AP_EFI_config.h"
+
+#if AP_EFI_MAV_ENABLED
+
+#include "AP_EFI.h"
```

Comment:
```suggestion
```

Reply:
As far as I know we still need this #include.
Or am I missing something?

Reply:
Once you're inheritting the constructor from the parent class you don't need to know the storage size of AP_EFI in this class.

You don't *actually* need the storage size of `AP_EFI` anyway, as you're getting a *reference* to the object; that means that you can just pre-declare `AP_EFI` as being a class an omit including the header.

Fewer headers == faster build times and, more importantly, faster *rebuild* times.

Reply:
So I should change
```
#include "AP_EFI.h"
#include "AP_EFI_Backend.h"
```
to
```
#include "AP_EFI_Backend.h"
class AP_EFI;
```
If I understand you correctly.

However, if do that, I get the following error while compiling:
![image](https://github.com/ArduPilot/ardupilot/assets/80035972/74aa69fc-a753-4f99-bd46-6c1205566eaf)

Reply:
Bleh.  Probably include loop.  Just leave the includes in place - we'll sort it out some other time :-)

The fix is *probably* to move the implemetation of `get_backend` into the `.cpp` file.

Reply:
Okay, I will leave both #includes for now.